test/distributed/org/apache/cassandra/distributed/impl/RMISslClientSocketFactoryImpl.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.distributed.impl;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.rmi.server.RMIClientSocketFactory;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.StringTokenizer;
+import javax.net.SocketFactory;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+
+import org.apache.cassandra.utils.RMICloseableSocketFactory;
+
+/**
+ * {@code RMIClientSocketFactory} for testing SSL based JMX clients.
+ * This class is used to override the local address the JMX client calculates when trying to connect,
+ * which can otherwise be influenced by the system property "java.rmi.server.hostname" in strange and
+ * unpredictable ways.
+ */
+public class RMISslClientSocketFactoryImpl implements RMIClientSocketFactory, Serializable,
+                                                      RMICloseableSocketFactory
+{
+    private static final long serialVersionUID = 9054380061905145241L;
+    private static final List<Socket> sockets = new ArrayList<>();
+    private final InetAddress localAddress;
+    private final String enabledCipherSuites;
+    private final String enabledProtocols;
+
+    public RMISslClientSocketFactoryImpl(InetAddress localAddress, String enabledCipherSuites, String enabledProtocls)
+    {
+        this.localAddress = localAddress;
+        this.enabledCipherSuites = enabledCipherSuites;
+        this.enabledProtocols = enabledProtocls;
+    }
+
+    @Override
+    public Socket createSocket(String host, int port) throws IOException
+    {
+        Socket socket = createSslSocket(port);
+        sockets.add(socket);
+        return socket;
+    }
+
+    private Socket createSslSocket(int port) throws IOException {
```

Comment:
@maulin-vasavada please put braces on new lines everywhere. Please be consistent with it.

Reply:
sorry about it. I forgot to reformat this.